src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joinSeparator = separator;
+
+    const lastIndex = this.length - 1;
+
+    let joinedString = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
```

Comment:
you don't need this check, because if there's no length - the loop just won't start iterating